File: app/src/main/java/com/thesunnahrevival/sunnahassistant/restapi/GeocodingInterface.java
package com.thesunnahrevival.sunnahassistant.restapi;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GeocodingInterface {

    @GET("maps/api/geocode/json")
    Call<GeocodingData> getGeocodingData(
            @Query("address") String address,
            @Query("key") String apiKey
    );

}

File: common/src/main/java/com/thesunnahrevival/common/data/remote/GeocodingInterface.kt
package com.thesunnahrevival.common.data.remote

import com.thesunnahrevival.common.data.model.GeocodingData
import retrofit2.http.GET
import retrofit2.http.Query

interface GeocodingInterface {
    @GET("maps/api/geocode/json")
    suspend fun getGeocodingData(
            @Query("address") address: String,
            @Query("key") apiKey: String,
            @Query("language") language: String
    ): GeocodingData?
}